File: Top_100_Liked_Questions/121._Best_Time_to_Buy_and_Sell_Stock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProf = 0
    let minPrice = Infinity;
    
    prices.forEach(price => {
        if (price < minPrice) {
            minPrice = price;
        } else {
            const profit = price - minPrice;
            if (profit > maxProf) {
                maxProf = profit;
            }
        }
    });
    
    return maxProf;
};

File: Top_100_Liked_Questions/121._Best_Time_to_Buy_and_Sell_Stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProf = 0
        minPrice = float('inf')
        
        for price in prices:
            if price < minPrice:
                minPrice = price
            else:
                maxProf = max(maxProf, price - minPrice)
        
        return maxProf